src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
I think it would be better to separate out the dependency from the metrics service in order to cut down the complexity of the error handling a bit.  One possibility might be to declare a `request-counter` atom inside of the `wrap-with-concurrent-request-limit` helper which is incremented/decremented as requests are created/finished.  That way, we'd have a local way in this new code path to determine the number of active requests.